nixos/modules/services/computing/boinc/client.nix

```diff
@@ -6,6 +6,24 @@ let
   cfg = config.services.boinc;
   allowRemoteGuiRpcFlag = optionalString cfg.allowRemoteGuiRpc "--allow_remote_gui_rpc";
 
+  # A package providing the boinc_client that will actually be executed.
+  execPkg = if cfg.useFHSEnv then
+    pkgs.buildFHSUserEnv {
+      name = "boinc_client";
+      targetPkgs = p: [ cfg.package ] ++ optional cfg.virtualbox.enable cfg.virtualbox.package;
+      multiPkgs = p: optional cfg.gpu.nvidia.enable cfg.gpu.nvidia.package;
+      runScript = "/bin/boinc";
+      extraBuildCommands = ''
+        if [ -e lib/vendors ]; then
+          chmod +w etc
+          mkdir -p etc/OpenCL
+          ln -s $(realpath lib/vendors) etc/OpenCL/vendors
+          chmod -w etc
+        fi
+      '';
```

Comment:
Thanks for the tip, I've updated the PR to use something along those lines.

I'll try and have a look to see how much (if anything) will break if we change the location of `nvidia.icd`, and how much work it would be to change that.